lib/aws.js

```diff
@@ -35,3 +35,53 @@ module.exports.getS3RVERConfiguration = function() {
         endpoint: new AWS.Endpoint('http://localhost:5000'),
     });
 };
+
+module.exports.uploadToS3 = async function (filePath, isDirectory, bucket, S3FilePath, localPath, callback) {
+    const s3 = new AWS.S3();
+
+    let body;
+    if (isDirectory) {
+        body = '';
+        S3FilePath += '/';
+    } else {
+        try {
+            body = await fs.promises.readFile(filePath);
+        } catch (err) {
+            callback(null, [filePath, S3FilePath, err]);
```

Comment:
I don’t like the way we return a length-3 array on error and a string “OK” on success. We should always return the same type. Also we should return errors normally as actual errors. We can capture the error in the caller if we want to suppress it.